In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
%matplotlib inline

In [3]:
df = pd.read_csv("data2.csv")
df.head()

address  area  baths  beds buildingtype   latitude  \
0        56 Walker Street #2  2670    3.5   3.0        Condo  40.719002   
1   201 East 62nd Street #5A  2240    3.5   3.0        Co-op  40.763500   
2    333 West 56th Street #2   820    1.0   1.0        Condo  40.766998   
3  330 East 38th Street #20P   884    1.5   1.0        Condo  40.746300   
4  322 West 57th Street #55U  1599    2.0   2.0        Condo  40.766800   

   longitude    neighborhood      price    borough    pricearea   coastdist  
0 -74.003197         Tribeca  5600000.0  manhattan  2097.378277  746.610278  
1 -73.964798      Lenox Hill  2595000.0  manhattan  1158.482143  501.933846  
2 -73.984901  Hell's Kitchen   995000.0  manhattan  1213.414634  598.449219  
3 -73.972702     Murray Hill  1225000.0  manhattan  1385.746606  257.252040  
4 -73.984200  Hell's Kitchen  4595000.0  manhattan  2873.671044  647.078718

In [4]:
df.tail()

address  area  baths  beds buildingtype   latitude  \
15293  11-02 49th Avenue #10N   737    1.0   1.0        Condo  40.742298   
15294   50-09 2nd Street #608  1186    1.5   1.0        Condo  40.742901   
15295   11-02 49th Avenue #3N   737    1.0   1.0        Condo  40.742298   
15296   11-02 49th Avenue #4D  1198    2.0   2.0        Condo  40.742298   
15297   120 Fifth Avenue #12F  3200    4.0   7.0        Condo  40.794102   

       longitude   neighborhood      price borough   pricearea   coastdist  
15293 -73.950996  Hunters Point   540000.0  queens  732.700136  355.071297  
15294 -73.957901  Hunters Point  1010000.0  queens  851.602024   76.061822  
15295 -73.950996  Hunters Point   450000.0  queens  610.583446  355.071297  
15296 -73.950996  Hunters Point   845000.0  queens  705.342237  355.071297  
15297 -73.825996          Malba   995000.0  queens  310.937500  116.533915

In [5]:
corr_matrix = df.corr()
corr_matrix['price'].sort_values(ascending=True)

longitude   -0.218206
coastdist   -0.191850
latitude     0.100591
beds         0.293536
area         0.528461
baths        0.561565
pricearea    0.742735
price        1.000000
Name: price, dtype: float64

In [6]:
print len(df.neighborhood.unique())
len(df.buildingtype.unique())

285


10

**We have 285 unique neighborhoods & 10 unique building types. What relevance will neighborhood have to the accuracy of our model? Some neighborhoods will definitely be highbrow and thus command higher price... What's the best way to encode the data in this column? Dummy variables? One-hot encoding?**